api/src/utils/channel-log-transformer/index.ts

```diff
@@ -0,0 +1,48 @@
+import { ChannelData, ChannelLog } from '../../models/types/channel-data';
+import { StreamsMessage } from '../../services/streams-service';
+
+export interface IPayload {
+	data?: any;
+	metadata?: any;
+	created?: any;
+	type?: string;
+}
+
+export class ChannelLogTransformer {
+	static transformStreamsMessages(data: StreamsMessage[]): ChannelData[] {
+		return data.map((data) => {
```

Comment:
To make sure it won't fail I would add optional chaining to the `data` array inself and all keys of the `data` object

Another point - here is a name clash: you use `data` as iterable, and the object inside is also named `data`